File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s283652523_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)
	scanner.Scan()
	N, _ := strconv.Atoi(scanner.Text())
	A := make([]int, N)
	for i := 0; i < N; i++ {
		scanner.Scan()
		num, _ := strconv.Atoi(scanner.Text())
		A[i] = num
	}
	const MOD = 1000000007 // Correct MOD declaration
	var ans int64
	var sum int64 = int64(A[N-1])
	for i := N - 2; i >= 0; i-- {
		prod := sum * int64(A[i])
		ans += prod % MOD
		ans %= MOD
		sum += int64(A[i])
		sum %= MOD
	}
	fmt.Println(ans % MOD)
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/inline/comment/direct/Java/Python/s283652523_commented.py
MOD = 10**9 + 7

def main():
    import sys
    data = sys.stdin.read().split()
    n = int(data[0])
    a = list(map(int, data[1:n+1]))
    ans = 0
    s = a[-1]
    for i in range(n-2, -1, -1):
        ans = (ans + (s * a[i]) % MOD) % MOD
        s = (s + a[i]) % MOD
    print(ans % MOD)

if __name__ == "__main__":
    main()
# End of Code
